### DISCUSSION TITLE
help/CI/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#17933] [Docs]
# help/CI/

Learn about Ultralytics CI actions, Docker deployment, broken link checks, CodeQL analysis, and PyPI publishing to ensure high-quality code.

https://docs.ultralytics.com/help/CI/

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for reaching out to Ultralytics 🚀! We appreciate your interest in learning more about our Continuous Integration (CI) workflows. For detailed information on CI actions and processes like Docker deployment, CodeQL analysis, and PyPI publishing, we recommend taking a look at our [CI Documentation](https://docs.ultralytics.com/help/CI/). 

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug and address the issue effectively.

If you're looking for help with a specific aspect of CI deployment or integration ❓, please provide more details about your use case, environment setup, and any error logs you're encountering to assist our team in understanding your request better.

Join the Ultralytics community for real-time discussions or to share feedback:
- Chat with us on [Discord](https://discord.com/invite/ultralytics) 🎧
- Join the conversation on [Discourse](https://community.ultralytics.com)
- Or explore threads on our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Ensure you're running the latest version of `ultralytics` by upgrading in your environment. This often resolves many issues as it includes the latest improvements and fixes:

```bash
pip install -U ultralytics
```

## CI Status

We recommend checking the current status of our CI runs. The badge below indicates whether all CI workflows are passing. CI tests are conducted on macOS, Windows, and Ubuntu across YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/).

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  
[Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule)

An Ultralytics engineer will assist you shortly. Thank you for your patience 😊!
